src/Scientist/Observation.cs

```diff
@@ -1,49 +1,178 @@
-﻿using GitHub.Internals;
+﻿using NullGuard;
 using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
 using System.Threading.Tasks;
 
 namespace GitHub
 {
     /// <summary>
-    /// Defines an observation of an experiment's execution.
+    /// Provides an interface for an observation of an experiment's execution.
     /// </summary>
-    public class Observation
+    public interface IObservation
```

Comment:
Ah I see!

I think the problem with the interfaces for publishing is we actually lose static type information that might be useful to an implementation of the publisher. So I tend to lean towards:

> Update the publishing interface to be `Task Publish<T>(Result<T> result)`

That way a single instance of a publisher can handle all publishing requests.